runtime/doc/lua.txt

```diff
@@ -4433,4 +4433,160 @@ tohtml.tohtml({winid}, {opt})                         *tohtml.tohtml.tohtml()*
+      • {scheme}?                   (`string`) (Added in 7.52.0)
+      • {size_download}?            (`number`)
+      • {size_header}?              (`number`)
+      • {size_request}?             (`number`)
+      • {size_upload}?              (`number`)
+      • {speed_download}?           (`number`)
+      • {speed_upload}?             (`number`)
+      • {ssl_verify_result}?        (`number`)
+      • {time_appconnect}?          (`number`)
+      • {time_connect}?             (`number`)
+      • {time_namelookup}?          (`number`)
+      • {time_pretransfer}?         (`number`)
+      • {time_redirect}?            (`number`)
+      • {time_starttransfer}?       (`number`)
+      • {time_total}?               (`number`)
+      • {url}?                      (`string`) (Added in 7.75.0)
+      • {url.scheme}?               (`string`) (Added in 8.1.0)
+      • {url.user}?                 (`string`) (Added in 8.1.0)
+      • {url.password}?             (`string`) (Added in 8.1.0)
+      • {url.options}?              (`string`) (Added in 8.1.0)
+      • {url.host}?                 (`string`) (Added in 8.1.0)
+      • {url.port}?                 (`string`) (Added in 8.1.0)
+      • {url.path}?                 (`string`) (Added in 8.1.0)
+      • {url.query}?                (`string`) (Added in 8.1.0)
+      • {url.fragment}?             (`string`) (Added in 8.1.0)
+      • {url.zoneid}?               (`string`) (Added in 8.1.0)
+      • {urle.scheme}?              (`string`) (Added in 8.1.0)
+      • {urle.user}?                (`string`) (Added in 8.1.0)
+      • {urle.password}?            (`string`) (Added in 8.1.0)
+      • {urle.options}?             (`string`) (Added in 8.1.0)
+      • {urle.host}?                (`string`) (Added in 8.1.0)
+      • {urle.port}?                (`string`) (Added in 8.1.0)
+      • {urle.path}?                (`string`) (Added in 8.1.0)
+      • {urle.query}?               (`string`) (Added in 8.1.0)
+      • {urle.fragment}?            (`string`) (Added in 8.1.0)
+      • {urle.zoneid}?              (`string`) (Added in 8.1.0)
+      • {urlnum}?                   (`number`) (Added in 7.75.0)
+      • {url_effective}?            (`string`)
+      • {xfer_id}?                  (`number`) (Added in 8.2.0)
+
+
+vim.net.download({url}, {opts})                           *vim.net.download()*
+    Asynchronously download a file
+
+    Parameters: ~
+      • {url}   (`string`) Request URL
+      • {opts}  (`table?`) Additional options
+
+                Example: >lua
+                    -- Download a file
+                    -- The file will be saved in the `cwd` with the name `anything`
+                    vim.net.download("https://httpbingo.org/anything")
+
+                    -- Download a file to a path
+                    -- The file will be saved in `/tmp/somefile`
+                    vim.net.download("https://httpbingo.org/anything", {
+                      download_location = "tmp/somefile",
```

Comment:
Maybe `as` or `to` or `target` would be shorter? People will specify this virtually always.

Reply:
Does this also support specifying a target _directory_? I think this would be useful, too.

Reply:
> Does this also support specifying a target directory? I think this would be useful, too.

Currently, no. A workaround may be to let users pass a `cwd` to execute `curl` there (and hence download the file on that directory)